### PR TITLE
Simple auto increment logic for item number

### PR DESCRIPTION
**Pull request by gammaw**
#14

See more details in the issue. https://github.com/mmmnmnm/lahmacun_arcsi/issues/14

### CODE REVIEW

File: arcsi/api/item.py [hunk 1]
```diff
@@ -91,9 +92,20 @@ def add_item():
     # work around ImmutableDict type
     item_metadata = request.form.to_dict()
     # TODO if we could send JSON payloads w/ ajax then this prevalidation isn't needed
+    show_id = int(item_metadata["shows"]) #shortcut in case the show id is needed in the next if statement
     item_metadata["shows"] = [
         {"id": item_metadata["shows"], "name": item_metadata["show_name"]}
     ]
+
+    if item_metadata["number"] == "":
+        show_request = requests.get(app.config["APP_BASE_URL"] + url_for("arcsi.view_show", id=show_id))
```
Comment (pvj) [inline arcsi/api/item.py:101]: I've read your comment in issue #14 and while it is true it doesn't have perf indication with our current usage we will definitely have to write this over so why not try it right first? This feels bit unnecessary. In my opinion, we have the `Show` object available to instantiate and could query & filter that which returns only the latest episode only if it has one, all in one.

File: arcsi/api/item.py [hunk 2]
```diff
@@ -91,9 +92,20 @@ def add_item():
...
+        show_dict = show_request.json()
+        if len(show_dict["items"])>0:
+            items_sorted = sorted(show_dict["items"], key=lambda k: k['number'], reverse = True)
```
Comment (pvj) [inline arcsi/api/item.py:104]: 👍

File: arcsi/templates/item/add.html [hunk 3]
```diff
@@ -30,7 +30,7 @@ <h2>{% block title %}Add episode to archive{% endblock %}</h2>
 </div>
 
   <div>
-    <label for="number">Episode number</label>
+    <label for="number">Episode number (+1 by def)</label>
```
Comment (pvj) [inline arcsi/templates/item/add.html:33]: I would try to word this differently. Leave empty to auto-increment or some such.

**Review by pvj — CHANGES_REQUESTED** [review]
Thanks for this, it's good code! I am still bit unsure if this is the right place for this indeed needed functionality and would advocate for something else. But since it is small change I think we could start w/ it living here and continue discussions further. But even, before we merge it I would suggest we use queries.

**Comment by gammaw** [comment]
I'm closing this, see details in https://github.com/mmmnmnm/lahmacun_arcsi/issues/14

**Pull request closed without merging**